src/renderer/types.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  RecordMatches,
+  RowBooleanOperations,
+  RowIDOperations,
+  RowNumberOperations,
+  RowStringOperations,
+  TableGenericOperations,
+  UnitTestOperations,
+} from 'db/entity/enum';
+
+export type Condition = {
+  level?: UnitTestOperations;
+  column?: string;
+  value?: string;
+  operation?:
+    | TableGenericOperations
+    | RowStringOperations
+    | RowBooleanOperations
+    | RowNumberOperations
+    | RowIDOperations;
+  result?: boolean;
+  expectedRecordMatches?: RecordMatches;
+  expectedNumRecords?: number;
+  total?: boolean;
+  table?: string;
+};
```

Comment:
Why do we have this type? Seems to just be a reduced version of the UnitTestType

Reply:
Think I see why you needed it but I don't think we need this type. 

We can use the built-in [`Partial`](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype) type which just makes all fields optional. This way we don't have to manage two types and keep them in sync

Reply:
It made my life easier.. not my proudest coding moment, and DEFINITELY is a top priority to figure out how to fix. I was running into a lot of annoyances with some of the values being required in the form. If you have a simple fix feel free to push.